File: Day5/AdventOfCode/Day1/Day1AdventOfCode.js
import fs from "fs/promises";

const START = 50;
const MAX = 100;

function zionPartOne(rotations) {
  let current = START;
  let password = 0;

  rotations.forEach((rotation) => {
    const line = rotation.trim();
    if (!line) return; // Skip empty lines

    const direction = line.slice(0, 1);
    const value = Number(line.slice(1));

    if (direction === "R") {
      current = (current + value) % MAX;
    } else if (direction === "L") {
      current = (current - value) % MAX;
      if (current < 0) {
        current = MAX + current;
      }
    }

    if (current === 0) {
      password += 1;
    }
  });

  console.log(password);
}

const input = await fs.readFile("../input.txt", "utf-8");
zionPartOne(input.split("\n"));

File: Day5/AdventOfCode/Day1/Day1AdventOfCode.py
START = 50
MAX_VAL = 100

def zion_part_one(file_path: str) -> int:
    current = START
    password = 0

    with open(file_path, "r", encoding="utf-8") as f:
        # .split() handles all whitespace (\n, \r, spaces) and skips empty lines automatically
        for rotation in f.read().split():
            direction = rotation[0]
            value = int(rotation[1:])

            if direction == 'R':
                current = (current + value) % MAX_VAL
            elif direction == 'L':
                current = (current - value) % MAX_VAL

            if current == 0:
                password += 1

    return password

if __name__ == "__main__":
    result = zion_part_one("../input.txt")
    print(result)